build/Build.Steps.cs

```diff
@@ -333,22 +335,21 @@ partial class Build
                 return;
             }
 
-            IEnumerable<TargetFramework> frameworks = IsWin ? TestFrameworks : TestFrameworks.ExceptNetFramework();
+            var frameworks = IsWin ? TestFrameworks : TestFrameworks.ExceptNetFramework();
 
             for (int i = 0; i < TestCount; i++)
             {
-                DotNetTest(config => config
+                DotNetMSBuild(config => config
                     .SetConfiguration(BuildConfiguration)
-                    .SetTargetPlatform(Platform)
+                    .SetPlatform(Platform)
```

Comment:
There is no way to pass Platform (x64/x86) through the dotnet tests to dotnet msbuild executed under the hood.
I have decided to call it directly.

This property is important, because .NET 7 SDK changed the output path structure based on this. Previously it was working only by accident.